component/azstorage/file_share.go

```diff
@@ -190,43 +191,183 @@ func (fs *FileShare) Exists(name string) bool {
 	return true
 }
 
+// CreateFile : Create a new file in the share/directory
 func (fs *FileShare) CreateFile(name string, mode os.FileMode) error {
-	return syscall.ENOTSUP
+	log.Trace("FileShare::CreateFile : name %s", name)
+
+	fileName, dirPath := getFileAndDirFromPath(filepath.Join(fs.Config.prefixPath, name))
+
+	fileURL := fs.Share.NewDirectoryURL(dirPath).NewFileURL(fileName)
+
+	_, err := fileURL.Create(context.Background(), 4398046511104, azfile.FileHTTPHeaders{
+		ContentType: getContentType(name),
+	},
+		nil)
+
+	if err != nil {
```

Comment:
print the error as well when you error log, so that we are aware of the reason for failure.